ext/standard/basic_functions.c

```diff
@@ -372,18 +372,18 @@ ZEND_BEGIN_ARG_INFO_EX(arginfo_array_splice, 0, 0, 2)
 	ZEND_ARG_INFO(0, replacement) /* ARRAY_INFO(0, arg, 1) */
 ZEND_END_ARG_INFO()
 
-ZEND_BEGIN_ARG_INFO_EX(arginfo_array_slice, 0, 0, 2)
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(arginfo_array_slice, 0, 2, IS_ARRAY, 0)
 	ZEND_ARG_INFO(0, arg) /* ARRAY_INFO(1, arg, 0) */
 	ZEND_ARG_INFO(0, offset)
 	ZEND_ARG_INFO(0, length)
 	ZEND_ARG_INFO(0, preserve_keys)
 ZEND_END_ARG_INFO()
 
-ZEND_BEGIN_ARG_INFO_EX(arginfo_array_merge, 0, 0, 0)
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(arginfo_array_merge, 0, 0, IS_ARRAY, 1)
```

Comment:
I've committed https://github.com/php/php-src/commit/4e391ee3a3086d4072012d5ed782517ed41ed400, so you can drop the nullability here and on array_merge_recursive now.